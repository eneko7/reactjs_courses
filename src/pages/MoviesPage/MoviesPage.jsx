import React from 'react';
import Header from '../../components/Header';
import ResultLine from '../../components/ResultLine';
import MoviesGrid from '../../components/MoviesGrid';
import Footer from '../../components/Footer';
import ErrorBoundary from '../../components/ErrorBoundary/ErrorBoundary';

const MoviesPage = () => (
  <ErrorBoundary>
    <Header />
    <ResultLine />
    <MoviesGrid />
    <Footer />
  </ErrorBoundary>
);

export default MoviesPage;
