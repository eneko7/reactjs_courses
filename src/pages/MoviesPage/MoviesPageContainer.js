import { connect } from 'react-redux';
import MoviesPage from './MoviesPage';
import { fetchSearchMovies } from '../../modules/movies/moviesActions';
import { getSearchBy, getSearchRequest, getSortBy } from '../../modules/filter/filterSelectors';

const mapStateToProps = state => ({
  searchBy: getSearchBy(state),
  sortBy: getSortBy(state),
  searchRequest: getSearchRequest(state),
});

const mapDispatchToProps = {
  fetchSearchMovies,
};
export default connect(mapStateToProps, mapDispatchToProps)(MoviesPage);
