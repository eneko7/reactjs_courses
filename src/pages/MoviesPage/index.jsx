import MoviesPage from './MoviesPageContainer';

export default MoviesPage;
