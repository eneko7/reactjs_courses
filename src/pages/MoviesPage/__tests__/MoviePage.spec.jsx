import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MoviesPage from '../MoviesPage';

const renderer = new ShallowRenderer();
describe('MoviesPage', () => {
  it('renders correctly', () => {
    renderer.render(<MoviesPage />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
