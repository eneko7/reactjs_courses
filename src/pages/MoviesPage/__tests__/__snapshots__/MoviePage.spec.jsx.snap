// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesPage renders correctly 1`] = `
<ErrorBoundary>
  <withRouter(Connect(Header)) />
  <withRouter(Connect(ResultLine)) />
  <Connect(MoviesGrid) />
  <Footer />
</ErrorBoundary>
`;
